File: qc-application/src/main/java/com/eager/questioncloud/application/api/authentication/AuthenticationToken.java
package com.eager.questioncloud.application.api.authentication;

import lombok.Getter;

@Getter
public class AuthenticationToken {
    private String accessToken;
    private String refreshToken;

    private AuthenticationToken(String accessToken, String refreshToken) {
        this.accessToken = accessToken;
        this.refreshToken = refreshToken;
    }

    public static AuthenticationToken create(String accessToken, String refreshToken) {
        return new AuthenticationToken(accessToken, refreshToken);
    }
}

File: qc-application/src/main/kotlin/com/eager/questioncloud/application/api/authentication/dto/AuthenticationToken.kt
package com.eager.questioncloud.application.api.authentication.dto

class AuthenticationToken(
    val accessToken: String,
    val refreshToken: String
) {
    companion object {
        fun create(accessToken: String, refreshToken: String): AuthenticationToken {
            return AuthenticationToken(accessToken, refreshToken)
        }
    }
}
